src/HtmlEntities.php

```diff
@@ -206,27 +58,30 @@ public function setDoubleQuote($doubleQuote)
      * If the value provided is non-scalar, the value will remain unfiltered
      *
      * @throws Exception\DomainException On encoding mismatches.
-     * @psalm-return ($value is scalar ? string : mixed)
      */
     public function filter(mixed $value): mixed
     {
-        if (! is_scalar($value)) {
+        if (! is_string($value) || $value === '') {
             return $value;
         }
         $value = (string) $value;
 
-        $filtered = htmlentities($value, $this->getQuoteStyle(), $this->getEncoding(), $this->getDoubleQuote());
-        if (strlen($value) && ! strlen($filtered)) {
+        $filtered = htmlentities($value, $this->quoteStyle, $this->encoding, $this->doubleQuote);
+        if (strlen($value) > 0 && strlen($filtered) === 0) {
```

Comment:
The value can no longer have zero length

```suggestion
        if (strlen($filtered) === 0) {
```